pandas/core/internals.py

```diff
@@ -442,12 +443,19 @@ def make_a_block(nv, ref_loc):
                     nv = _block_shape(nv, ndim=self.ndim)
                 except (AttributeError, NotImplementedError):
                     pass
-                block = self.make_block(values=nv,
-                                        placement=ref_loc, fastpath=True)
+
```

Comment:
I don't like this pattern at all, nv *should* be a sparse array at this point; the inference should just work. is that not the case?